### PR TITLE
content type handling for custom error

### PR DESCRIPTION
**Pull request by bakiss21**
More flexible content type handling for the custom error functionality.

### CODE REVIEW

File: server_response.go [hunk 1]
```diff
@@ -172,8 +171,22 @@ func SendProblemResponse(w http.ResponseWriter, r *http.Request, statusCode int,
 	return
 }
 
+func acceptContentType(r *http.Request, contentType string) bool {
+	if !strings.HasPrefix(contentType, "application/") {
```
Comment (Som-Som-CC) [inline server_response.go:175]: What happens, if client sent `Accept: text/plain` and server calls `NewErrorWithBody(nil, http.StatusBadRequest, "text/plain", "invalid field")`?

File: server_response.go [hunk 2]
```diff
@@ -172,8 +171,22 @@ func SendProblemResponse(w http.ResponseWriter, r *http.Request, statusCode int,
...
+		return false
+	}
+	acceptHeader := r.Header.Values(AcceptHeader)
```
Comment (Som-Som-CC) [inline server_response.go:178]: `acceptHeaders` in plural sounds better.

File: server_response.go (hunk 1 above)
Comment (Som-Som-CC) [reply server_response.go:175]: I'd check exact match with base content type here. Or go on if `Accept` header is missing completely.

Comment (bakiss21) [reply server_response.go]: sendCustomResponse and acceptContentType functions are updated

**Pull request merged**